mlir/lib/Dialect/Tosa/IR/TosaOps.cpp

```diff
@@ -2675,6 +2674,139 @@ LogicalResult TransposeConv2DOp::verify() {
   return success();
 }
 
+LogicalResult RescaleOp::verify() {
+  auto inputType = llvm::dyn_cast<ShapedType>(getInput().getType());
+  if (!inputType || !inputType.hasRank()) {
```

Comment:
Ah, if the operator doesn't have a rank, it could still be considered valid, until it can be proven it's invalid - I think it's safer to `return success()` in this case. This way we avoid later checks that assume the type is ranked, but allow other transformation passes to infer/update the type

Reply:
Done